pkg/distributor/distributor.go

```diff
@@ -876,74 +900,87 @@ func detectLogLevelFromLogEntry(entry logproto.Entry, structuredMetadata labels.
 		} else if otlpSeverityNumber <= int(plog.SeverityNumberFatal4) {
 			return logLevelFatal
 		}
-		return logLevelInfo
+		return logLevelUnknown
 	}
 
 	return extractLogLevelFromLogLine(entry.Line)
 }
 
 func extractLogLevelFromLogLine(log string) string {
-	// check for log levels in known log formats to avoid any false detection
+	var v string
+	if isJSON(log) {
+		v = getValueUsingJSONParser(log)
+	} else {
+		v = getValueUsingLogfmtParser(log)
+	}
+
+	switch strings.ToLower(v) {
+	case "trace", "trc":
+		return logLevelTrace
+	case "debug", "dbg":
+		return logLevelDebug
+	case "info", "inf":
+		return logLevelInfo
+	case "warn", "wrn":
+		return logLevelWarn
+	case "error", "err":
+		return logLevelError
+	case "critical":
+		return logLevelCritical
+	case "fatal":
+		return logLevelFatal
+	default:
+		return detectLevelFromLogLine(log)
+	}
+}
+
+func getValueUsingLogfmtParser(line string) string {
+	equalIndex := strings.Index(line, "=")
+	if len(line) == 0 || equalIndex == -1 {
```

Comment:
cc @benclive do you think we could have something more resilient ?

Reply:
Unfortunately not. According to the formal definition, basically everything is valid logfmt. That means that this filter can discard valid logfmt log lines if none of the keys had values. See [this fluentd issue](https://github.com/fluent/fluent-bit/issues/4927) for some discussion.
I reality, I think anything using logfmt would have at least one key=value pair so this is probably the best you can do to validate/guess.
https://pkg.go.dev/github.com/kr/logfmt#pkg-overview contains the definition.